src/util/cmdline.h

```diff
@@ -39,7 +39,9 @@ class cmdlinet
 protected:
   struct optiont
   {
-    bool isset, hasval, islong;
+    bool isset=false;
+    bool hasval=false;
+    bool islong=false;
     char optchar;
```

Comment:
May I suggest a default constructor doing this work, and then also initialising `optchar` (to 0)?